bundler/lib/dependabot/bundler/update_checker/version_resolver.rb

```diff
@@ -77,50 +75,42 @@ def fetch_latest_resolvable_version_details
           return latest_version_details unless gemfile
 
           SharedHelpers.with_git_configured(credentials: credentials) do
-            in_a_temporary_bundler_context do
-              dep = dependency_from_definition
-
-              # If the dependency wasn't found in the definition, but *is*
-              # included in a gemspec, it's because the Gemfile didn't import
-              # the gemspec. This is unusual, but the correct behaviour if/when
-              # it happens is to behave as if the repo was gemspec-only.
-              if dep.nil? && dependency.requirements.any?
-                next latest_version_details
-              end
-
-              # Otherwise, if the dependency wasn't found it's because it is a
-              # subdependency that was removed when attempting to update it.
-              next nil if dep.nil?
-
-              # If the dependency is Bundler itself then we can't trust the
-              # version that has been returned (it's the version Dependabot is
-              # running on, rather than the true latest resolvable version).
-              next nil if dep.name == "bundler"
-
-              # If the old Gemfile index was used then it won't have checked
-              # Ruby compatibility. Fix that by doing the check manually (and
-              # saying no update is possible if the Ruby version is a mismatch)
-              next nil if ruby_version_incompatible?(dep)
-
-              details = { version: dep.version }
-              if dep.source.instance_of?(::Bundler::Source::Git)
-                details[:commit_sha] = dep.source.revision
+            in_a_native_bundler_context(error_handling: false) do |tmp_dir|
```

Comment:
Any reason this isn't handling errors like before?

```suggestion
            in_a_native_bundler_context do |tmp_dir|
```

Reply:
Yeah, in this case there are a few errors the version resolver wants to handle first, if it doesn't handle them, it manually hands off to `handle_bundler_errors` on line [105](https://github.com/dependabot/dependabot-core/pull/2600/files/419713075e150d03c506c8f98fd29cde43f0f2f3#diff-3092b190ebf8948a50c8d770607f2417R105)

I was doing it differently up until [this commit](https://github.com/dependabot/dependabot-core/pull/2600/commits/419713075e150d03c506c8f98fd29cde43f0f2f3) but having the rescues inside the block lead to having to manage the temp manifest files being re-written after dropping a `.gemspec` which was hard to follow.